docs/analysing-compilation-output.rst

```diff
@@ -0,0 +1,190 @@
+.. index:: analyse, asm
+
+#############################
+Analysing the Compiler Output
+#############################
+
+It is often useful to look at the assembly code generated by the compiler. However, the generated
+binary, i.e., the output of ``solc --bin contract.sol``, is difficult to read. It is recommended to use
+the flag ``--asm`` to analyse the assembly output.
```

Comment:
Maybe add: Even for large contracts, looking a a graphical diff of the assembly before and after a change is often very enlightening.